src/assets.psm1

```diff
@@ -2126,3 +2126,807 @@ function Remove-SafeguardAssetAccount
+        {
+            $local:Args = @{
+                AccessToken = $AccessToken
+                Appliance = $Appliance
+                Insecure = $true
+                ParentAsset = $local:Account.ParentAsset
+                NewAccountName = $local:Account.NewAccountName
+            }
+
+            if($null -ne $local:Account.Description) 
+            {
+                $local:Args.Add("Description", $local:Account.Description)
+            }
+
+            if($null -ne $local:Account.AssetPartition) 
+            {
+                $local:Args.Add("AssetPartition", $local:Account.AssetPartition)
+            }
+
+            if($null -ne $local:Account.DomainName) 
+            {
+                $local:Args.Add("DomainName", $local:Account.DomainName)
+            }
+            
+            if($null -ne $local:Account.DistinguishedName) 
+            {
+                $local:Args.Add("DistinguishedName", $local:Account.DistinguishedName)
+            }
+
+            New-SafeguardAssetAccount @local:Args
+        }
+        catch 
+        {
+            if ($local:Account.PSobject.Properties.Name -contains "Error")
+            {
+                $local:Account.Error = $_
+            }
+            else 
+            {
+                $local:Account | Add-Member -MemberType NoteProperty -Name "Error" -Value  $_
+            }
+            $local:FailedImports.Add($local:Account)
+        }
+        
+        Write-Progress -Activity "Importing Asset Accounts ..." -PercentComplete (($local:CurrAccount/$local:Accounts.Count)*100)
+        $local:CurrAccount++
+    }
+
+    Write-Host ($local:Accounts.Count - $local:FailedImports.Count) "Successful Imports," $local:FailedImports.Count "Failed Imports"
+    
+    if ($local:FailedImports.Count -gt 0) 
+    {
+        Write-Host "Please refer to AssetAccountImportResults.csv for more information on failures."
+        $local:FailedImports | Export-Csv -Path ".\AssetAccountImportResults.csv" -NoTypeInformation -Force
+    }
+}
+
+<#
+.SYNOPSIS
+Creates a template file containing the headers for importing assets.
+
+.DESCRIPTION
+Creates a template file containing the headers for importing assets.
+
+Columns
+
+- AssetPartition : An integer containing an ID or a string containing the name of the asset partition to set the asset account password in.
+
+- AssetToSet : An integer containing the ID of the asset to set account password on or a string containing the name.
+
+- AccountToSet : An integer containing the ID of the account to set password on or a string containing the name.
+
+- NewPassword : A string containing the new password to set.
+
+.INPUTS
+None.
+
+.OUTPUTS
+A CSV file with the headers.
+
+.EXAMPLE
+New-SafeguardAssetAccountPasswordImportTemplate
+
+#>
+function New-SafeguardAssetAccountPasswordImportTemplate
+{
+    $local:Path = '.\SafeguardAssetAccountPasswordImportTemplate.csv';
+
+    $local:Headers = '"AssetPartition","AssetToSet","AccountToSet","NewPassword"'
+
```

Comment:
Can these be parameters with default values? I think it's better to use:

Join-Path $PSScriptRoot 'SafeguardAssetAccountPasswordImportTemplate.csv'
or
Join-Path (pwd) 'SafeguardAssetAccountPasswordImportTemplate.csv'

To be explicit about where the file will be created. I also think this is a good reason for making this a parameter in case the caller wants it to go somewhere else.

Reply:
Added Path parameter for all ...Template cmdlets